src/mango/test/26-test_query.py

```diff
@@ -0,0 +1,150 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+import copy
+import mango
+import unittest
+import json
+
+BOOLEN_ADB_DOC = {
+    "_id": "hoary_marmot",
+    "wiki_page": "http://en.wikipedia.org/wiki/Hoary_marmot",
+    "min_weight": 4,
+    "max_weight": 40,
+    "min_length": 0.2,
+    "max_length": 1,
+    "latin_name": "Marmota caligata",
+    "class": "mammal",
+    "diet": "herbivore",
+    "inherit_the_earth": True,
+    "likes": ["to take tea", "kittens", "trees"],
+    "habitat": [{"mountainous_environments": 8200, "states": ["Alaska", "Canada"]}],
+}
+
+BOOLEN_ADB_DOC2 = {
+    "_id": "hoary_marmots",
+    "wiki_page": "http://en.wikipedia.org/wiki/Hoary_marmot",
+    "min_weight": 3,
+    "max_weight": 30,
+    "min_length": 0.2,
+    "max_length": 1,
+    "latin_name": "Marmota caligatas",
+    "class": "mammal",
+    "diet": "carnivore",
+    "inherit_the_earth": True,
+    "likes": ["to take tea", "kittens", "trees", "games"],
+    "habitat": [{"mountainous_environments": 8200, "states": ["Alaska", "Canada"]}],
+}
+
+KOOKABURRA = {
+    "_id": "kookaburra",
+    "min_length": 0.28,
+    "max_length": 0.42,
+    "wiki_page": "http://en.wikipedia.org/wiki/Kookaburra",
+    "class": "bird",
+    "diet": "carnivore",
+    "latin_name": "Dacelo novaeguineae",
+}
+
+DDOC = {
+    "_id": "_design/views101",
+    "indexes": {
+        "animals": {
+            "index": """
+                function(doc){
+                  index("default", doc._id);
+                  if(doc.min_length){
+                    index("min_length", doc.min_length, {"store": "true"});
+                  }
+                  if (doc['class']){
+                    index("class", doc['class'], {"store": "true"});
+                  }
+                }
+            """
+        }
+    },
+}
+
+DOCS = [BOOLEN_ADB_DOC, BOOLEN_ADB_DOC2, KOOKABURRA, DDOC]
+
+
+@unittest.skipUnless(mango.has_text_service(), "requires text service")
+class TestQuery(mango.DbPerClass):
+    def setUp(self):
+        self.db.recreate()
```

Comment:
This is wrong. It recreates the database with the same name. Which is makes the test flaky.

Reply:
This is unfortunately a common pattern in the Mango tests.  Perhaps the implementation of `recreate` shall be fixed to avoid that.